### DISCUSSION TITLE
Is it possible to filter out rows by a related model's id?

### DISCUSSION THREAD

**Discussion by talonx** [prisma/prisma#23693] [Q&A] [answered]
### Question

The question's title is not the best, so I'll try to explain it here. 
Models -
```prisma
model Provider {
  id             Int              @id
  name           String
  url            String           @unique
  logo           String
  desc           String
  statusPage     String
  ProviderConfig ProviderConfig[]
}

model UserChannel {
  id               Int             @id @default(autoincrement())
  name             String
  type             String
  userid           String
  data             String /// @encrypted
  desc             String
  // user             User            @relation(fields: [userid], references: [id])
  providerConfig   ProviderConfig? @relation(fields: [providerConfigId], references: [id])
  providerConfigId Int?
}

model ProviderConfig {
  id        Int           @id @default(autoincrement())
  userid    String        @unique
  // user      User          @relation(fields: [userid], references: [id])
  providers Provider[]
  channels  UserChannel[]
}

```

Given the url of a Provider, I want to fetch all the ProviderConfig which have the Provider in them, and also return the Channel[] from those configs. Fetching the provider is trivial but I'm stuck in the next step. 
```js
    let p = await prismadb.provider.findUnique({
        where: {
            url: url
        }
    });
```
To fetch channels
```js
    let channels = await prismadb.providerConfig.findMany({
        where: {
            providers: {
                     ///filter based on p.id (from the previous query) being present as one of the Provider[] in this model.
            }
        },
        include: {
            channels: true,
        }
    });
```
Is this possible? Any help appreciated.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @talonx 👋 

It is possible to filter out rows by a related model's `id` in Prisma. You can use the `some` keyword in the where clause to filter based on the existence of a related record. See this section on [filtering by relations](https://www.prisma.io/docs/orm/prisma-client/queries/filtering-and-sorting#filter-on-relations)
```js
let channels = await prisma.providerConfig.findMany({
  where: {
      providers: {
        some:{
          id: p.id
        }
      }
  },
  include: {
      channels: true,
  }
});
```
In this query, `some` checks if there is at least one `Provider` related to the ProviderConfig that matches the condition `(id: p.id)`. If there is, the `ProviderConfig` is included in the result set. The `include` clause then includes the related `Channel` records in the result.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by talonx** [reply]
Thank you, that worked.
